### ISSUE TITLE
proposal: `empty_control_statement`

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#4321]
## Description

Control statements that are empty often indicate that the author forgot something. They should either be deleted or filled with content. (Or, in the rare case where they are supposed to be empty, a comment explaining why they are empty should be added.)

## Kind

Guard against errors

## Bad Examples

```dart
void foo(int a) {
  for (int i = 1; i < 10; i++) {
  }
  
  if (a == 1) { 
  } else {
  }
  
  if (a == 2) {
  }
}
```

## Good Examples 

```dart
void foo(int a) {
  for (int i = 1; i < 10; i++) {
     // busy waiting.
  }
  
  if (a == 1) { 
    print('A is one');
  } else {
    print('A is not one');
  }
  
  if (a == 2) {
    print('A is two');
  }
}
```

## Discussion

I was running into some empty control statements (that according to the author were accidentally left over from debugging) and was surprised that there was no lint to remove those, see https://github.com/flutter/flutter/pull/125437/files/454fae6be148c14b44e17e684c007dd8a0985a7b#diff-ea66aca7737cb104f299e8a21e1a6d8125ab69c45d2dfa326f1efeb1cefa1f02.

### Discussion checklist

- [x] List any existing rules this proposal modifies, complements, overlaps or conflicts with.
- [x] List any relevant issues (reported here, the [SDK Tracker], or elsewhere).
- [x] If there's any prior art (e.g., in other linters), please add references here.
- [x] If this proposal corresponds to [Effective Dart] or [Flutter Style Guide] advice, please call it out. (If there isn't any corresponding advice, should there be?)
- [x] If this proposal is motivated by real-world examples, please provide as many details as you can.  Demonstrating potential impact is _especially valuable_.

[Writing Lints]: https://github.com/dart-lang/linter/blob/main/doc/writing-lints.md
[Effective Dart]: https://dart.dev/guides/language/effective-dart
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[SDK Tracker]: https://github.com/dart-lang/sdk/issues
